game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -207,6 +209,46 @@ public Territory getTerritory(final String s) {
     return getNeighbors(frontier, searched, distance, null);
   }
 
+  /**
+   * Returns all neighbors within a certain distance of the starting territory that territories
+   * between the 2 match the condition. Does NOT include the original/starting territory in the
+   * returned Set.
+   */
+  public Set<Territory> getNeighborsIgnoreEnd(final Territory territory, final int distance,
+      final Predicate<Territory> cond) {
+    if (distance < 0) {
+      throw new IllegalArgumentException("Distance must be positive not: " + distance);
```

Comment:
really nit-picking, 'zero' is not a positive number and distnace can be zero.

Suggestion, as a one-liner with guava:
```
Preconditions.checkArgument(distance < 0, "Distance must be non-negative: " + distance);
```

Reply:
Updated.